### ISSUE TITLE
Contribution and unit tests

### ISSUE THREAD

**Issue by anjmao** [primefaces/primeng#3116]
**I'm submitting a ...**  (check one with "x")
```
[ ] bug report => Search github for a similar issue or PR before submitting
[x] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35
```

This seems like a good library with a lot of components, but is it possible to contribute to this library by fixing issues? If yes then why I don't see any unit tests and Travis build? Does authors of this library care about quality? I am a bit confused, how can I use this library in production or contribute to it knowing what there are not unit tests? :)

**Comment by cagataycivici** [comment]
Please use forum for discussions not the issue tracker, the forum has topics about this and we've discussed it there e.g.

https://forum.primefaces.org/viewtopic.php?f=35&t=51358

**Comment by benburton** [comment]
The referenced forum topic was mysteriously deleted after [I pointed out on twitter](https://twitter.com/bjburton/status/921435896767045632) that there weren't any unit tests. @cagataycivici said there were unit tests "after alpha" in a since deleted tweet:

<img width="303" alt="screen shot 2017-10-22 at 4 04 32 pm" src="https://user-images.githubusercontent.com/168757/31865816-dbf61166-b742-11e7-9944-b11f61145c7e.png">

...and then the thread referenced above was deleted. Thankfully, we have a [Google cache](https://webcache.googleusercontent.com/search?q=cache:9nj5k6lAKZwJ:https://forum.primefaces.org/viewtopic.php%3Ft%3D51358+&cd=1&hl=en&ct=clnk&gl=us) of the conversation.

The rhetoric, from what I can tell, is as follows:

1. Angular was upgraded
2. Unit tests didn't work with the new version of Angular.
3. We deleted all the unit tests.
4. Maybe we'll re-introduce the unit tests at some point in the future

There are no unit tests in this repository. The last note that it was being addressed was from June, and it's now October (almost November). 5 months and no unit tests. Well-tested code is clearly not a priority for this library.

I am taking steps to disparage anyone on our team from using PrimeNG. The premise that having well-tested code is secondary to shipping is antithetical to any organization that wishes to have a well-functioning product.

Without tests, PrimeNG is a bad library. It should not be used by anyone who is serious about the  stability of their application.
